pandas/tests/io/parser/test_dtypes.py

```diff
@@ -550,3 +550,21 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+@pytest.mark.parametrize("na_string", ["NaN", "nan", "NA", "null", "NULL", ""])
+@pytest.mark.parametrize("true_string", ["True", "TRUE", "true"])
+@pytest.mark.parametrize("false_string", ["False", "FALSE", "false"])
+def test_boolean_dtype(all_parsers, na_string, true_string, false_string):
```

Comment:
This parametrization generates a ton of test cases which increases pytest collection / run time. Can you inline all of them in a single test?

```python
data = "\n".join(["a", "true", "TRUE", "True", "false", "FALSE", "False", "NaN", "nan", ...])
```

and then make an assert?

Reply:
That makes a lot more sense than what I was doing. Regarding the assert, something like

```
assert all([s in data for s in ["True", "TRUE", "true"]])
assert all([s in data for s in ["False", "FALSE", "false"]])
assert all([s in data for s in ["NaN", "nan", "NA", "null", "NULL"]])
```

(separating them out to make a failure more meaningful)? I dropped the empty string for now because it seems to get ignored in a single column CSV.